### PR TITLE
feat: APP-163 custodial wallet design implementation

### PR DESCRIPTION
**Pull request by r41ph**
## Description

https://regennetwork.atlassian.net/browse/APP-163

This PR also fixes an error with missing providers in the related stories in Storybook: https://regen-storybook.netlify.app/?path=/story/registry-organisms-useraccountsettings--default  

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [x] provided a link to the relevant issue or specification
- [x] provided instructions on how to test
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed

### How to test

1. https://deploy-preview-2470--regen-storybook.netlify.app/?path=/story/marketplace-organisms-useraccountsettings--width-560
2. https://deploy-preview-2470--regen-storybook.netlify.app/?path=/story/marketplace-organisms-useraccountsettings--width-375

~~Note: the `Learn more»` link needs to be updated with the correct URL~~

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by r41ph** [comment]
@erikalogie see testing instructions

**Comment by erikalogie** [comment]
LGTM!

File: web-storybook/.storybook/preview.jsx [hunk 1]
```diff
@@ -31,14 +38,20 @@ const StorybookI18nProvider = ({ children }) => {
...
-    <StorybookI18nProvider>
-      <ThemeProvider injectFonts>
-        <CssBaseline />
-        <Router>
-          <Story />
-        </Router>
-      </ThemeProvider>
-    </StorybookI18nProvider>
+    <QueryClientProvider client={reactQueryClient}>
+      <AuthApolloProvider apolloClientFactory={apolloClientFactory}>
```
Comment (blushi) [inline web-storybook/.storybook/preview.jsx:42]: not sure if we should have storybook depend on those providers, which are really marketplace related
Instead if we want to keep the UserAccountSettings story, we should probably refactor it to not use those providers directly but rather provide what we need as props (from `useEmailConfirmationData`)

**Review by blushi — COMMENTED** [review]
looks good, just one suggestion

**Review by blushi — APPROVED** [review]
tACK

**Pull request merged**